rbi/core/kernel.rbi

```diff
@@ -1916,7 +1916,7 @@ module Kernel
     params(
         arg0: BasicObject,
     )
-    .returns(NilClass)
+    .returns(BasicObject)
```

Comment:
Maybe we can use a `type_parameter` to return the same type as the argument.

Reply:
Ah interesting, something like this?

```ruby
  sig do
    type_parameters(:T)
      .params(arg0: T.type_parameter(:T))
      .returns(T.type_parameter(:T))
  end
```

Reply:
Yeah, exactly that.